pkg/plugin/state.go

```diff
@@ -630,13 +634,24 @@ func (e *AutoscaleEnforcer) reserveResources(
 
 	e.state.lock.Lock()
 	defer e.state.lock.Unlock()
+	podName := util.GetNamespacedName(pod)
 
 	// If the pod already exists, nothing to do
-	if _, ok := e.state.pods[util.GetNamespacedName(pod)]; ok {
+	_, isPodInState := e.state.pods[podName]
+	if isPodInState {
 		logger.Info("Pod already exists in global state")
 		return true, &verdictSet{cpu: "", mem: ""}, nil
 	}
 
+	// log the pods if:
+	// - pod is assigned to autoscaler scheduler
+	// - pod not in the state
+	// - pod is not preexisting pod
+	// - pod has the node name
+	if !isPodInState && !opts.preexisting && pod.Spec.SchedulerName == autoscalerSchedulerName && pod.Spec.NodeName != "" {
+		logger.Warn("Pod is not in the state and scheduling metainformation is inconsistent")
```

Comment:
> scheduling metainformation is inconsistent

It's less about inconsistency and more that: If the pod doesn't exist in the state, _but_ we were supposed to schedule it, then it means we were bypassed for the scheduling of this pod.